src/wasm-binary.h

```diff
@@ -562,44 +618,53 @@ class WasmBinaryWriter : public WasmVisitor<void> {
     for (auto target : curr->targets) {
       o << mapping[target];
     }
-    visit(curr->value);
+    breakStack.push_back(curr->name);
+    recurse(curr->value);
     for (auto c : curr->cases) {
-      visit(c.body);
+      recurse(c.body);
     }
+    breakStack.pop_back();
   }
   void visitCall(Call *curr) {
+    if (debug) std::cerr << "zz node: Call" << std::endl;
     o << int8_t(BinaryConsts::CallFunction) << LEB128(getFunctionIndex(curr->target));
     for (auto operand : curr->operands) {
-      visit(operand);
+      recurse(operand);
     }
   }
   void visitCallImport(CallImport *curr) {
+    if (debug) std::cerr << "zz node: CallImport" << std::endl;
     o << int8_t(BinaryConsts::CallFunction) << LEB128(getFunctionIndex(curr->target));
     for (auto operand : curr->operands) {
-      visit(operand);
+      recurse(operand);
     }
   }
   void visitCallIndirect(CallIndirect *curr) {
-    o << int8_t(BinaryConsts::CallFunction) << LEB128(getFunctionTypeIndex(curr->fullType->name));
+    if (debug) std::cerr << "zz node: CallIndirect" << std::endl;
+    o << int8_t(BinaryConsts::CallIndirect) << LEB128(getFunctionTypeIndex(curr->fullType->name));
+    recurse(curr->target);
     for (auto operand : curr->operands) {
-      visit(operand);
+      recurse(operand);
     }
   }
   void visitGetLocal(GetLocal *curr) {
+    if (debug) std::cerr << "zz node: GetLocal " << (o.size() + 1) << std::endl;
     o << int8_t(BinaryConsts::GetLocal) << LEB128(mappedLocals[curr->name]);
   }
   void visitSetLocal(SetLocal *curr) {
+    if (debug) std::cerr << "zz node: SetLocal" << std::endl;
     o << int8_t(BinaryConsts::SetLocal) << LEB128(mappedLocals[curr->name]);
-    visit(curr->value);
+    recurse(curr->value);
   }
 
   void emitMemoryAccess(size_t alignment, size_t bytes, uint32_t offset) {
-    o << int8_t( ((alignment == bytes || alignment == 0) ? 0 : 128) |
-                 (offset ? 8 : 0) );
+    o << int8_t( ((alignment == bytes || alignment == 0) ? 0 : BinaryConsts::Alignment) |
```

Comment:
An alignment of `0` in the binary format means natural alignment? If so, it would be good to make it a symbolic constant instead.